### PR TITLE
feat: Accept Invite Home Page Changes

### PR DESCRIPTION
**Pull request by PritishBudhiraja**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/HyperSwitch/SwitchMerchant/SwitchMerchant.res [hunk 1]
```diff
@@ -146,41 +123,36 @@ module ExternalUser = {
...
+      setSelectedMerchantObject(_ => extractMerchantObject)
     }
 
-    React.useEffect1(() => {
-      fetchMerchantIDs()->ignore
+    React.useEffect2(() => {
+      fetchMerchantIDs()
       None
-    }, [merchantDetailsTypedValue.merchant_name])
+    }, (merchantDetailsTypedValue.merchant_name, switchMerchantList))
```
Comment (JeevaRamu0104) [inline src/screens/HyperSwitch/SwitchMerchant/SwitchMerchant.res]: please check the length of `switchMerchantList` in the useEffect

File: src/entryPoints/hyperswitch/AcceptInviteHome.res [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  let (merchantListValue, setMerchantListValue) = React.useState(_ => [])
+  let (acceptedMerchantId, setAcceptedMerchantId) = React.useState(_ => [])
+
+  React.useEffect1(() => {
+    let filteredSwitchMerchantList = switchMerchantListValue->Array.filter(ele => !ele.is_active)
+    setMerchantListValue(_ => filteredSwitchMerchantList)
+    let arr = Array.make(~length=filteredSwitchMerchantList->Array.length, false)
+    setAcceptedMerchantId(_ => arr)
+    None
+  }, [switchMerchantListValue])
```
Comment (JeevaRamu0104) [inline src/entryPoints/hyperswitch/AcceptInviteHome.res:18]: can you please check the length of switchMerchantListValue

File: src/entryPoints/hyperswitch/AcceptInviteHome.res [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        ]->LogicUtils.getJsonFromArrayOfJson
+      let _ = await updateDetails(url, body, Post, ())
+      showToast(~toastType=ToastSuccess, ~message="Invite Accepted Successfully", ())
+    } catch {
+    | _ => ()
+    }
+    setShowModal(_ => false)
+  }
+
+  switch merchantListValue->Array.length {
```
Comment (JeevaRamu0104) [inline src/entryPoints/hyperswitch/AcceptInviteHome.res:47]: this switch case looks bit wired can we go with  
<UIUtils.RenderIf condition={merchantListValue->Array.length==1}>
<UIUtils.RenderIf condition={merchantListValue->Array.length>1}>

File: src/screens/HyperSwitch/SwitchMerchant/SwitchMerchant.res (hunk 1 above)
Comment (PritishBudhiraja) [reply src/screens/HyperSwitch/SwitchMerchant/SwitchMerchant.res]: Updated

File: src/entryPoints/hyperswitch/AcceptInviteHome.res (hunk 2 above)
Comment (PritishBudhiraja) [reply src/entryPoints/hyperswitch/AcceptInviteHome.res:18]: Updated

File: src/entryPoints/hyperswitch/AcceptInviteHome.res (hunk 3 above)
Comment (PritishBudhiraja) [reply src/entryPoints/hyperswitch/AcceptInviteHome.res:47]: Updated

**Review by JeevaRamu0104 — APPROVED** [review]

**Pull request merged**